app/src/main/java/com/fer1592/k8s_android_console/view/fragments/ClustersFragment.kt

```diff
@@ -77,17 +80,39 @@ class ClustersFragment : Fragment() {
             }
         }
 
+        // Sets observer to show progress bar when deleting a cluster
+        clustersViewModel.processingData.observe(viewLifecycleOwner) {
+            if (it) showLoading()
+            else hideLoading()
+        }
+
         // Set on click listener for the FAB to add a new cluster
         binding.addCluster.setOnClickListener {
             val action = ClustersFragmentDirections.actionClustersFragmentToSetClusterFragment(-1)
             this.findNavController().navigate(action)
         }
 
+        showLoading()
         return view
     }
 
     override fun onDestroyView() {
         super.onDestroyView()
         _binding = null
     }
+
+    override fun onResume() {
+        super.onResume()
+        binding.clustersList.scheduleLayoutAnimation()
+    }
+
+    private fun showLoading() {
+        (activity as MainActivity).binding.mainProgressBar.visibility = View.VISIBLE
+        (activity as MainActivity).binding.navHostFragment.visibility = View.GONE
+    }
+
+    private fun hideLoading() {
```

Comment:
this method should be defined inside the MainActivity as public

Reply:
done!